ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/results/TransactionCompleteResult.java

```diff
@@ -99,10 +98,14 @@ public TransactionCompleteResult(final TransactionWithMetadata tx) {
     this.from = transaction.getSender().toString();
     this.gas = Quantity.create(transaction.getGasLimit());
     this.maxPriorityFeePerGas =
-        tx.getTransaction().getMaxPriorityFeePerGas().map(q -> q.toShortHexString()).orElse(null);
+        tx.getTransaction().getMaxPriorityFeePerGas().map(Wei::toShortHexString).orElse(null);
     this.maxFeePerGas =
-        tx.getTransaction().getMaxFeePerGas().map(q -> q.toShortHexString()).orElse(null);
-    this.gasPrice = transaction.getGasPrice().map(Quantity::create).orElse(null);
+        tx.getTransaction().getMaxFeePerGas().map(Wei::toShortHexString).orElse(null);
+    this.gasPrice =
+        Quantity.create(
+            transaction
+                .getGasPrice()
+                .orElse(Wei.of(transaction.getEffectivePriorityFeePerGas(tx.getBaseFee()))));
```

Comment:
I think this should be `getEffectiveGasPrice` , not `getEffectivePriorityFeePerGas`

https://github.com/ethereum/execution-specs/pull/251